sklearn/linear_model/_logistic.py

```diff
@@ -2016,15 +2028,54 @@ def score(self, X, y, sample_weight=None):
         sample_weight : array-like of shape (n_samples,), default=None
             Sample weights.
 
+        **score_params : dict
+            Parameters to pass to the `score` method of the underlying scorer.
+
         Returns
         -------
         score : float
             Score of self.predict(X) wrt. y.
         """
-        scoring = self.scoring or "accuracy"
-        scoring = get_scorer(scoring)
+        scoring = self._get_scorer()
+        routed_params = process_routing(
+            obj=self,
+            method="score",
+            sample_weight=sample_weight,
+            other_params=score_params,
+        )
+
+        return scoring(
+            self, X, y, sample_weight=sample_weight, **routed_params.scorer.score
+        )
 
-        return scoring(self, X, y, sample_weight=sample_weight)
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self(self)
+            .add(
+                splitter=self.cv,
+                method_mapping=MethodMapping().add(callee="split", caller="fit"),
+            )
+            .add(
+                scorer=self._get_scorer(),
+                method_mapping=MethodMapping()
+                .add(callee="score", caller="score")
+                .add(callee="score", caller="fit"),
```

Comment:
fit is not passing extra params to scorer at the moment. I think that's missing

Reply:
Do you mean we should pass the params to the _log_reg_scoring_path?

Reply:
yes.